openeo/rest/connection.py

```diff
@@ -677,16 +677,20 @@ def _build_request_with_process_graph(self, process_graph: dict, **kwargs) -> di
     # TODO: Maybe rename to execute and merge with execute().
     def download(self, graph: dict, outputfile):
```

Comment:
if `outputfile` can be string/Path or `None` I would add typehint and default here:
`outputfile: Union[Path, str, None] = None`